### ISSUE TITLE
Sending Data From An Async Generator To A Websocket

### ISSUE THREAD

**Issue by iwoloschin** [tiangolo/fastapi#106]
**Description**
I'm attempting to stream a text log file through a websocket (think `tail -f` but through a browser).  I have a working async generator that appears to read my file's contents in near realtime, but it appears that the `await websocket.send_json(data)` doesn't do anything until the async generator ends, and then everything that has been queued up gets dumped to the websocket.  Is this expected behavior?  If it is expected, are there any ways to work around this?

```python
async for data in read_log(log_filename):
    log.critical(data)
    await websocket.send_json({"data": data})
```

In my FastAPI log I see data printed in realtime, but nothing appears on the other end of the websocket until `read_log` returns (instead of yields).  After `read_log` returns, all of the log file's contents are dumped to the websocket and appear on the other side as expected.  I was hoping to get data to show up as it's printed into the log file, same as you'd expect with `tail -f`.

**Additional context**
The intended use case here is to simulate a `tail -f` in an Angular SPA, but it can be tested by using the `websocat` tool (https://github.com/vi/websocat).

Also, this is my first foray into async python, so I could just be doing everything completely wrong!  I'm pretty sure I'm not...but I really have no idea what I'm doing here :).

**Comment by euri10** [comment]
maybe dumb question but are you sure `read_log(log_filename)` is async ?
from memory aiofiles is a nice library to use for that.

**Comment by iwoloschin** [comment]
Nope, not a stupid question, looks like that was the problem.  I was thrown off since I was seeing the right data in my console log so I assumed it was doing the right thing.  Looks like `aiofiles` does the right thing though, thanks!

**Comment by tiangolo** [comment]
Great to see you solved your problem @iwoloschin.

You rock @euri10! Thanks for all your help here.

**Comment by CurryGaifan** [comment]
How to solve this problem please. @iwoloschin

**Comment by CurryGaifan** [comment]
I solved this problem. I used `time.sleep` incorrectly in async method.
The code below can work.

```
async def get_text():
    for i in range(5):
        yield i
        # time.sleep(2)  
        await asyncio.sleep(5)

async for text in get_text():
    await websocket.send_text(f"Message text was: {text}")
    logger.info(text)
```
